### DISCUSSION TITLE
How to use relative Path in quarto ?

### DISCUSSION THREAD

**Discussion by ghost** [quarto-dev/quarto-cli#9201] [Q&A] [answered]
### Description

Hi ! 

I want to make a template for a quarto presentation, and then, I want to make a "Tools" folder outside the presentation's folder, like that :

````qmd
---
format:
  revealjs: 
    slide-number: true
    chalkboard: 
      buttons: true
    preview-links: auto
    logo: ./../tools/images/image.png
    css: ./../tools/styles.css
---
````

I am getting the 404:Not found Error.

Could you tell me how should I do for that ?

Thanks !

**Comment by mcanouil** [comment, marked as answer]
> I want to make a "Tools" folder outside the presentation's folder

All resources should be contained within your Quarto project, that's a design choice. (A project root is defined either by the document or by `_quarto.yml`).
Paths are resolved using `/` as root of the project, not as root of the file system.

**Reply by ghost** [reply]
I'm revisiting the documentation available at https://quarto.org/docs/projects/quarto-projects.html.

My current scenario involves embedding presentations within my repository. I'm exploring the possibility of creating a configuration file at the root of the repository to streamline the creation of Quarto presentations. With this approach, I aim to maintain consistency across all presentations by utilizing the styles and logo provided with absolute paths.

Could you confirm if this workflow is feasible within Quarto?

**Reply by ghost** [reply]
Not sur I well understoof the purpose of _quarto.yml

Now that I made one as the template, the outputs are in the _output folder, which is pretty cool because I can add it to git ignore.

But for exemple, the styles.css, changed with the correct path, does not change the style of my presentation. But now the relative PATH works.  I attend to use quarto more often, and wanted to understand why. 

Thanks a lot !

**Reply by mcanouil** [reply]
Without knowing/having the structure, we can't help more or explain what you are seeing.
You need to share a Git repository for example.

> My current scenario involves embedding presentations within my repository. I'm exploring the possibility of creating a configuration file at the root of the repository to streamline the creation of Quarto presentations. With this approach, I aim to maintain consistency across all presentations by utilizing the styles and logo provided with absolute paths.
> 
> Could you confirm if this workflow is feasible within Quarto?

That's the whole purpose of defining a Quarto project.
See <https://quarto.org/docs/projects/quarto-projects.html#shared-metadata> regarding `_quarto.yml`.

**Reply by ghost** [reply]
Sorry, I attended to ask for a general purpose : if I am working in a git repo, where I attend to put many .qmd, does _quarto.yml is relevant as a general configuration file in order that all .qmd s could search by exemple the same style.css and the logo (in order to not have to copy paste every time and make the repo too heavy) ?

**Reply by ghost** [reply]
This confusing for me, because I understand that the .yml is just for one project - one presentation.
If I attend to make a git repo template where people are saving many presentations .qmd, and having one style, basics image, in order to use the same quarto template, by saving at the same time the more space that I can.

**Reply by mcanouil** [reply]
> Sorry, I attended to ask for a general purpose 

Your question is not general as it relies on your current *unknown* directory structure.

>  if I am working in a git repo, where I attend to put many .qmd, does _quarto.yml is relevant as a general configuration file in order that all .qmd

Did you read the link I shared? https://quarto.org/docs/projects/quarto-projects.html#shared-metadata

-> "All Quarto projects include a `_quarto.yml` configuration file. Any document rendered within the project directory will automatically inherit the metadata defined at the project level."

Is it not clear?


> search by exemple the same style.css and the logo (in order to not have to copy paste every time and make the repo too heavy)

This is clearly a Quarto extension you are describing here. See <https://quarto.org/docs/extensions/>.

**Reply by ghost** [reply]
Hi ! 

Thank you for your answer !

Yes I read and I used it for my repo : I was confused if it says "you should have for any project a .yml" or "any project could have a single .yml for configuration". But for many presentations, only having works so it's good for me.

Thank you again for you time and explonations !
